docs/js/settings.js

```diff
@@ -184,25 +184,46 @@ const UserSettings = {
         return this._custom_colors_on;
     },
 
-    _local_storage: 1,
+    // This setting is for whether the user wants local storage to be used at all, ever
+    _local_storage: true,
     set local_storage(newValue) {
-        const valueInt = newValue ? parseInt(newValue, 10) : 1;
-        this._local_storage = valueInt;
-
-        document.getElementById("clear_storage_opt").value = valueInt;
-        switch (valueInt) {
-            case 1:
-                deleteCookie('local_storage');
-                break;
-            case 4:
-                setCookie('local_storage', valueInt, 2147483647);
-                break;
+        let valueInt;
+        if (typeof newValue === 'boolean') {
+            valueInt = newValue ? 1 : 4;
+        } else {
+            valueString = String(newValue);
+
+            if (valueString.match(/[14]/)) {
+                valueInt = parseInt(newValue, 10);
+            } else if (valueString.match(/true|false/i)) {
+                valueInt = valueString === 'true' ? 1 : 4;
+            }
         }
+        
+        this._local_storage = (valueInt === 1);
+        
+        document.getElementById("allow_local_storage").value = valueInt;
+        this.attemptSave();
     },
     get local_storage() {
         return this._local_storage;
     },
 
+    // This setting is for whether the current puzzle should be saved
+    _save_current_puzzle: undefined,
+    set save_current_puzzle(newValue) {
+        this._save_current_puzzle = newValue;
+    },
+    get save_current_puzzle() {
+        // Check if we explicitly set this puzzle to not allow saving to local storage.
+        if (this._save_current_puzzle !== undefined) {
+            return this._save_current_puzzle;
+        }
+        
+        // Default to whether saving is turned on in general.
+        return (this._local_storage === 1);
```

Comment:
I think this should be
`return this._local_storage`

Because it is a boolean value and comparing it with === 1 always return false and the puzzle is never saved.